File: 1991.kt
class Solution {
    fun findMiddleIndex(nums: IntArray): Int {
    val sumAll = nums.sum()
    var sumLeft = 0
    for (i in nums.indices) {
        if (sumLeft == sumAll - sumLeft - nums[i]) {
            return i
        }
        sumLeft += nums[i]
    }
    return -1
    }
}

File: 1991.rs
impl Solution {
    pub fn find_middle_index(nums: Vec<i32>) -> i32 {
    let sum_all: i32 = nums.iter().sum();
    let mut sum_left = 0;
    for (i, &num) in nums.iter().enumerate() {
        if sum_left == sum_all - sum_left - num {
            return i as i32;
        }
        sum_left += num;
    }
    -1
    }
}
